src/action.js

```diff
@@ -6,59 +6,62 @@ const registerDependencyAndroid = require('./android/registerNativeModule');
 const registerDependencyIOS = require('./ios/registerNativeModule');
 const copyAssetsAndroid = require('./android/copyAssets');
 const copyAssetsIOS = require('./ios/copyAssets');
+const pjson = require(path.join(process.cwd(), './package.json'));
 
+/**
+ * Namespace log messages
+ */
 log.heading = 'rnpm-link';
 
 /**
  * Returns an array of dependencies that should be linked/checked.
+ * @param {Object} pconfig Project config
  */
-const getProjectDependencies = () => {
-  const pjson = require(path.join(process.cwd(), './package.json'));
-  return Object.keys(pjson.dependencies).filter(name => name !== 'react-native');
-};
+const getProjectDependencies = (pconfig) =>
```

Comment:
This is quite misleading because in below method `pconfig` refers to result from `getProjectConfig` but we receive `pjson` here.